gui/RespiraConnectedDevice.h

```diff
@@ -0,0 +1,115 @@
+#include "../common/generated_libs/network_protocol/network_protocol.pb.h"
+#include "../common/third_party/nanopb/pb_common.h"
+#include "../common/third_party/nanopb/pb_decode.h"
+#include "../common/third_party/nanopb/pb_encode.h"
+#include "connected_device.h"
+#include <QSerialPort>
+#include <QThread>
+
+#define INTER_FRAME_TIMEOUT 11
+
+class RespiraConnectedDevice : public ConnectedDevice {
+
+public:
+  RespiraConnectedDevice(QString portName, uint32_t write_timeout_ms,
+                         uint32_t interframe_timeout_ms)
+      : serialPortName(portName), m_writeTimeout_ms(write_timeout_ms),
+        m_interframeTimeout_ms(interframe_timeout_ms) {}
+
+  ~RespiraConnectedDevice() {
+    if (m_serial != nullptr) {
+      m_serial->close();
+    }
+  }
+  void createPort() {
+    if (m_serial != nullptr)
+      return;
+    printf("Socket create\n");
+    m_serial = new QSerialPort();
+    m_serial->setPortName(serialPortName);
+    m_serial->setBaudRate(QSerialPort::Baud115200);
+    m_serial->setDataBits(QSerialPort::Data8);
+    m_serial->setParity(QSerialPort::NoParity);
+    m_serial->setStopBits(QSerialPort::OneStop);
+    m_serial->setFlowControl(QSerialPort::NoFlowControl);
+    if (!m_serial->open(QIODevice::ReadWrite)) {
+      printf("Could not open serial port %s\n",
```

Comment:
In QT, you can include `<QtDebug>` and log like this: `qCritical() << "Could not open serial port " << serialPortName;`
There's also qDebug, qInfo, qFatal (crashes if anything is logged to it).